src/org/labkey/test/pages/pipeline/PipelineStatusDetailsPage.java

```diff
@@ -76,7 +77,16 @@ public String getEmail()
 
     public String getStatus()
     {
-        return elementCache().statusText.getText();
+        String retVal = null;
+        try
+        {
+            retVal = elementCache().statusText.getText();
+        }
+        catch (StaleElementReferenceException e)
+        {
+            return retVal;
+        }
+        return retVal;
```

Comment:
I don't think this will work the way you want it to.

For example, looking at the call stack PipelineStatusDetailsPage is called which has this line:
`        waitFor(() -> FINISHED_STATES.contains(getStatus()), () -> "Pipeline job did not finish. Final state: " + getStatus(), timoutMs);`

if `getStatus` returns null then `FINISHED_STATES.contains(getStatus())` will throw a null pointer exception.

Since this function will now possibly return a null you will need to look at all the callers of this method and decide if they need to be able to deal with a null.

I think to accomplish what you want you should change PipelineStatusDetailsPage to:
`        waitFor(() -> (null != getStatus()) && FINISHED_STATES.contains(getStatus()), () -> "Pipeline job did not finish. Final state: " + getStatus(), timoutMs);`

Reply:
That is not correct. `FINISHED_STATES` could, theoretically, contain `null`. The check is fine as is.

Reply:
D'oh, yep you are right @labkey-tchad  and I was wrong. I thought FINISHED_STATES was a string.